sycl/include/CL/sycl/detail/common.hpp

```diff
@@ -118,7 +118,7 @@ static inline std::string codeToString(cl_int code) {
 #define __SYCL_REPORT_OCL_ERR_TO_EXC(expr, exc)                                \
   {                                                                            \
     auto code = expr;                                                          \
-    if (code != CL_SUCCESS) {                                                  \
+    if (code != CL_SUCCESS && code != -998) {                                  \
```

Comment:
I think this change should not be needed. This new code is not a success. If caller of piextGetDeviceFunctionPointer is comfortable with this return code they should check it manually like you did in sycl/source/function_pointer.cpp

Reply:
Yes, I will undo it and change the the call to `call_nocheck` and handle the codes from the caller.